logrotate.c

```diff
@@ -1454,6 +1458,7 @@ static int prerotateSingleLog(struct logInfo *log, int logNum,
     if (log->compress_ext && (log->flags & LOG_FLAG_COMPRESS))
         compext = log->compress_ext;
 
+    localtime_r(&nowSecs, &now);
```

Comment:
Could we optimize out the `now` variable and write to `state->lastRotated` directly?

Reply:
I do not think so; `now` is later used in the function https://github.com/logrotate/logrotate/blob/5e7cecb677e4a35aa77423d7aff9304061327013/logrotate.c#L1510

Reply:
Indeed.  My bad.